ui/app/AppLayouts/Browser/popups/BrowserConnectionModal.qml

```diff
@@ -42,164 +41,159 @@ Popup {
+                    width: 20
+                    height: 4
                 }
-                if (root.currentAddress === selectedAccount.address) {
-                    return
+
+                RoundedIcon {
+                    source: Style.svg("check")
+                    iconColor: Style.current.primary
+                    color: Style.current.secondaryBackground
+                    width: 24
+                    height: 24
                 }
 
-                root.currentAddress = selectedAccount.address
-                Web3ProviderStore.web3ProviderInst.dappsAddress = selectedAccount.address;
-                Web3ProviderStore.revokeAllPermissions()
+                SVGImage {
+                    id: dots2
+                    source: Style.svg("dots-icon")
+                    width: 20
+                    height: 4
+                }
 
-                if (selectedAccount.address) {
-                    Web3ProviderStore.web3ProviderInst.dappsAddress = selectedAccount.address;
-                    WalletStore.setDappBrowserAddress()
+                RoundedIcon {
+                    source: Style.svg("walletIcon")
+                    iconHeight: 18
+                    iconWidth: 18
+                    iconColor: accountSelector.selectedAccount.iconColor || Style.current.primary
+                    color: Style.current.background
+                    width: logoHeader.imgSize
+                    height: logoHeader.imgSize
+                    border.width: 1
+                    border.color: Style.current.border
                 }
             }
-        }
 
+            StatusBaseText {
+                id: titleText
+                //% "'%1' would like to connect to"
+                text: qsTrId("--1--would-like-to-connect-to").arg(request.title)
+                wrapMode: Text.WordWrap
+                font.weight: Font.Bold
+                font.pixelSize: 17
+                horizontalAlignment: Text.AlignHCenter
+                elide: Text.ElideRight
+                Layout.maximumWidth: browserConnectionModal.width - Style.current.padding
+                Layout.fillWidth: true
+                Layout.alignment: Qt.AlignHCenter | Qt.AlignTop
+            }
 
-        StyledText {
-            id: infoText
-            text: {
-                switch(request.permission){
-                //% "Allowing authorizes this DApp to retrieve your wallet address and enable Web3"
-                case Constants.permission_web3: return qsTrId("allowing-authorizes-this-dapp");
-                //% "Granting access authorizes this DApp to retrieve your chat key"
-                case Constants.permission_contactCode: return qsTrId("your-contact-code");
-                default: return qsTr("Unknown permission: " + request.permission);
+            StatusAccountSelector {
+                id: accountSelector
+                label: ""
+                width: 300
+                implicitWidth: width
```

Comment:
i think you can set `implicitWidth` directly without `width`

Reply:
yep!